File: apps/admin_mobile/lib/application/use_cases/reports/get_range_report_use_case.dart
import '../../../domain/repositories/report_repository.dart';

class GetRangeReportUseCase {
  final ReportRepository _repo;
  GetRangeReportUseCase(this._repo);

  Future<Map<String, dynamic>> call(String from, String to) =>
      _repo.getRange(from, to);
}

File: apps/admin-v2/src/application/use-cases/reports/get-range-report.use-case.ts
import type { ReportRepository, RangeReportFilters } from '@/domain/repositories/report.repository';

export class GetRangeReportUseCase {
  constructor(private repo: ReportRepository) {}

  execute(from: string, to: string, filters?: RangeReportFilters) {
    return this.repo.getRange(from, to, filters);
  }
}
